deepmd/train/trainer.py

```diff
@@ -443,15 +447,18 @@ def _init_sess_distrib(self):
         # ,
         # save_checkpoint_steps = self.save_freq)
 
-    def train (self, 
-               data) :
+    def train (self, train_data, valid_data=None) :
+
+        if valid_data is None:  # no validation set specified.
+            valid_data = train_data  # using training set as validation set.
```

Comment:
The setting `valid_data = train_data` may not be a good option. It gives the user the wrong impression that their model is validated and the validate error looks almost the same as the training error. 

If `valid_data` is `None`, I would suggest not printing the column of valid error in `lcurve.out`.